### PR TITLE
refactoring: introducing rollupjs

### PR DESCRIPTION
**Pull request by julianpoemp**
I really like this project! After trying to use it in one of my Angular apps I was facing the issue that I can't import the withLineNumbers module (see #61). I was curious why it doesn't work and did a deep dive in this project. After I found some issues I did some refactoring on top of a build tool: [rollup.js](https://rollupjs.org/guide/en/)

My changes have some advantages:

- The project is bundled easily using `npm run build`. It creates an IIFE and an ES bundle automatically
  - The IIFE bundle is available in two versions: An minified one and a unminified one.
  - It adds a header to the bundles containing the package information (like name, version, build time etc) automatically, e.g.:
    ````
    /**
     * codejar v3.4.0
     * Generated: 2021-06-09 01:16:17
     * Author: Anton Medvedev <anton@medv.io>
     * LICENSE: MIT
     */
    ````
- Rollupjs does have a lot more plugins to improve building bundles.
- The ES build can be imported easily in any Typescript project.
- More object oriented programming. CodeJar is now an Object that can be initialized with `new CodeJar`. Because I don't know if my solution will be accepted I did only necessery changes.

I added a test build to this pull request:

[codejar-rollup.zip](https://github.com/antonmedv/codejar/files/6620156/codejar-rollup.zip)

### CODE REVIEW

**Comment by antonmedv** [comment]
Awesome! Thanks for your contribution. But lets separate it into independent parts.

**Comment by julianpoemp** [comment]
@antonmedv what do you mean with "lets seperate it into independent parts"?

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,752 @@
-/node_modules/
-package-lock.json
-*.js
-*.d.ts
+### Eclipse ###
```
Comment (antonmedv) [inline .gitignore:1]: This is a really big file. What is purpose of it? Let’s not add it.

Comment (julianpoemp) [reply .gitignore:1]: the package.lock.json should be added to repositories. See https://stackoverflow.com/a/44210813/6303600

**Comment by julianpoemp** [comment]
@antonmedv if you would decide to use this change of project structure you could ignore this pull request and copy the new files and folders manually to the main branch and commit it. Bacause I did only some refactoring I don't need to be added as contributor. The most of these changes are complete replacements.

**Comment by antonmedv** [comment]
Let's try to discuss such changes first in an issue and integrate by small PRs. This is too big to merge.

**Comment by julianpoemp** [comment]
Hi @antonmedv what do you think about this issue? Using rollup.js would solve issue #61, too.

**Pull request closed without merging**